contentgrid-spring-data-support/src/test/java/com/contentgrid/spring/data/support/auditing/v1/JwtAuditorAwareTest.java

```diff
@@ -0,0 +1,198 @@
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.data.querydsl.QuerydslPredicateExecutor;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+import org.springframework.data.rest.core.config.RepositoryRestConfiguration;
+import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurer;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.JwtRequestPostProcessor;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.servlet.config.annotation.CorsRegistry;
+
+@SpringBootTest
+@ContextConfiguration
+class JwtAuditorAwareTest {
+
+    private MockMvc mockMvc;
+
+    @Autowired
+    InvoiceRepository invoices;
+
+    @Autowired
+    WebApplicationContext context;
+
+    @BeforeEach
+    public void setup() {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(context)
+                .apply(springSecurity())
+                .alwaysDo(print())
+                .build();
+    }
+
+    @AfterEach
+    public void cleanup() {
+        invoices.deleteAll();
+    }
+
+    static JwtRequestPostProcessor jwtWithClaims(String subject, String name) {
+        return jwt().jwt(jwt -> jwt.subject(subject).claim("name", name)
+                .issuer("http://localhost/realms/cg-invalid"));
+    }
+
+    static JwtRequestPostProcessor john = jwtWithClaims("123456", "john smith");
+    static JwtRequestPostProcessor bob = jwtWithClaims("abcdef", "bob admin");
+
+    @Test
+    void postEntity_shouldSetAuditMetadataFields_http201() throws Exception {
+        mockMvc.perform(post("/invoices")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {"number": "123456"}
+                                """)
+                        .with(john))
+                .andExpect(status().isCreated());
+        var dateAfterCreation = Instant.now();
+
+        mockMvc.perform(get("/invoices").with(john))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$._embedded.['d:invoices'][0].number").value("123456"))
```

Comment:
Doesn't mockmvc have a lenient json matcher ? That would help readability a lot!